repo2docker/__main__.py

```diff
@@ -181,6 +198,13 @@ def get_argparser():
         action="store_true",
         help="Push docker image to repository",
     )
+    argparser.add_argument(
+        "--no-push",
+        dest="push",
+        action="store_false",
+        help="Don't push docker image to repository (default).",
+    )
+    argparser.set_defaults(push=None)
```

Comment:
`--no-clean` and `--no-push` differ in a way I don't understand, by having `argparser.set_defaults(push=None)` compared to having `default=None,`. Is this intented or does it make sense to do it the same way?

Reply:
They aren't different on purpose, I just didn't implement both directions for every boolean flag. Failing to implement both directions only means one case isn't covered: when config inverts the default for a bool, there's no flag for setting it back to the default on a per-call basis. This isn't _strictly_ required, but it seems relevant to cover for at least some flags, hence the `--[no-]run`, etc.

The reason for `argparser.set_defaults(push=None)` is that there are now _two_ arguments defining the `push` argument, so it's ambiguous which one should set the default (or maybe both need to?), so setting the default outside the two argument definitions seemed to me to be the clearest.

Reply:
Ah, makes sense to have it set once outside both of them then - good point!